lib/mac/highsierra.js

```diff
@@ -0,0 +1,403 @@
+
+blenoBindings.updateRssi = function() {
+  if (this._deviceUUID === null) {
+    this.emit('rssiUpdate', 127); // not supported
+  } else {
+    this.sendCBMsg(61, {
+      kCBMsgArgDeviceUUID: this._deviceUUID
+    });
+  }
+};
+
+blenoBindings.on('kCBMsgId29', function(args) {
+  var attributeId = args.kCBMsgArgAttributeID;
+  var result = args.kCBMsgArgResult;
+
+  if (result) {
+    var errorMessage = 'failed to set service ' + this._attributes[attributeId].uuid;
+
+    if (result === 27) {
+      errorMessage += ', UUID not allowed!';
+    }
+
+    this._setServicesError = new Error(errorMessage);
+  }
+
+  if (attributeId === this._lastServiceAttributeId) {
+    this.emit('servicesSet',  this._setServicesError);
+  }
+});
+
+blenoBindings.on('kCBMsgId57', function(args) {
+  var deviceUUID = args.kCBMsgArgDeviceUUID.toString('hex');
+  var mtu = args.kCBMsgArgATTMTU;
+
+  this._deviceUUID = new Buffer(deviceUUID, 'hex');
+  this._deviceUUID.isUuid = true;
+
+  uuidToAddress(deviceUUID, function(error, address) {
+    this.emit('accept', address);
+    this.emit('mtuChange', mtu);
+  }.bind(this));
+});
+
+blenoBindings.on('kCBMsgId30', function(args) {
+  var attributeId = args.kCBMsgArgAttributeID;
+  var offset = args.kCBMsgArgOffset || 0;
+  var transactionId = args.kCBMsgArgTransactionID;
+
+  var callback = (function(attributeId, transactionId) {
+    return function(result, data) {
+      this.sendCBMsg(21, {
+        kCBMsgArgAttributeID: attributeId,
+        kCBMsgArgData: data,
+        kCBMsgArgResult: result,
+        kCBMsgArgTransactionID: transactionId
+      });
+    }.bind(this);
+  }.bind(this))(attributeId, transactionId);
+
+  this._attributes[attributeId].emit('readRequest', offset, callback);
+});
+
+blenoBindings.on('kCBMsgId31', function(args) {
+  var attWrites = args.kCBMsgArgATTWrites;
+  var transactionId = args.kCBMsgArgTransactionID;
+
+  for (var i = 0; i < attWrites.length; i++) {
+    var attWrite = attWrites[i];
+
+    var attributeId = attWrite.kCBMsgArgAttributeID;
+    var data = attWrite.kCBMsgArgData;
+    var ignoreResponse = attWrite.kCBMsgArgIgnoreResponse ? true : false;
+    var offset = args.kCBMsgArgOffset || 0;
+
+    var callback = (function(attributeId, transactionId, ignoreResponse) {
+      return function(result) {
+        if (!ignoreResponse) {
+          this.sendCBMsg(21, {
+            kCBMsgArgAttributeID: attributeId,
+            kCBMsgArgData: null,
+            kCBMsgArgResult: result,
+            kCBMsgArgTransactionID: transactionId
+          });
+        }
+      }.bind(this);
+    }.bind(this))(attributeId, transactionId, ignoreResponse);
+
+    this._attributes[attributeId].emit('writeRequest', data, offset, ignoreResponse, callback);
+  }
+});
+
+blenoBindings.on('kCBMsgId32', function(args) {
+  var attributeId = args.kCBMsgArgAttributeID;
+  var maxValueSize = 20;
+
+  var callback = (function(attributeId) {
+    return function(data) {
+      this.sendCBMsg(15, {
```

Comment:
Not too sure about this. Not seeing a reactive call to xpc 15 in my test app. Anyone have a good test case?

Reply:
I believe this should be `22` instead of 15. 
With 22 I'm getting notifications on my central.